src/Widgets/HumbleButton.vala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2016 elementary LLC (https://launchpad.net/granite)
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+public class AppCenter.Widgets.HumbleButton : Gtk.Grid {
+    public signal void download_requested ();
+    public signal void payment_requested (int amount);
+
+    private Gtk.Popover selection;
+    private Gtk.Button amount_button;
+    private Gtk.SpinButton custom_amount;
+
+    private Gtk.ToggleButton arrow_button;
+
+    private int amount_;
+    public int amount {
+        get {
+            return amount_;
+        } set {
+            amount_ = value;
+            amount_button.label = get_amount_formated (value, true);
+            custom_amount.value = value;
+
+            if (amount != 0) {
+                amount_button.label = get_amount_formated (amount, true);
+            } else {
+                amount_button.label = _("Free");
+            }
+        }
+    }
+
+    public string label {
+        set {
+            amount_button.label = value;
+        }
+    }
+
+    public bool can_purchase {
+        set {
+            if (!value) {
+                amount = 0;
+            }
+
+            arrow_button.visible = value;
+            arrow_button.no_show_all = !value;
+        }
+    }
+
+    public bool sugested_action {
```

Comment:
I think it would be better here if you could remove these styles by setting the `suggested_action` to `false` just to be consistent.